src/log_surgeon/finite_automata/RegexNFA.hpp

```diff
@@ -39,6 +54,34 @@ class RegexNFAState {
         return m_matching_variable_id;
     }
 
+    auto add_positive_tagged_transition(
+            uint32_t const tag,
+            RegexNFAState<state_type> const* dest_state
+    ) -> void {
+        m_positive_tagged_transitions.push_back(
+                PositiveTaggedTransition<state_type>{tag, dest_state}
+        );
+    }
+
+    [[nodiscard]] auto get_positive_tagged_transitions(
+    ) const -> std::vector<PositiveTaggedTransition<state_type>> const& {
+        return m_positive_tagged_transitions;
+    }
+
+    auto add_negative_tagged_transition(
+            std::set<uint32_t> const& tags,
```

Comment:
```suggestion
            std::set<uint32_t> tags,
```
I think we should use modern pass-by-value with `std::move`

Reply:
Changed.

Reply:
Changed.